tutorials/api/phasorpy_multi-harmonic.py

```diff
@@ -0,0 +1,459 @@
+fluorescence lifetime and hyperspectral images with the PhasorPy library.
+
+"""
+
+# %%
+# Import required modules and functions:
+
+import numpy
+import tifffile  # TODO: from phasorpy.io import read_ometiff
+from matplotlib import pyplot
+
+from phasorpy.datasets import fetch
+
+# %%
+# Read signal from file
+# ---------------------
+#
+# Throughout this tutorial, we will use an Imspector TIFF file from the
+# `FLUTE <https://zenodo.org/records/8046636>`_  project containing a
+# time-correlated single photon counting (TCSPC) histogram
+# of a zebrafish embryo at day 3, acquired at 80 MHz.
+#
+# Read the signal from the file:
+
+
+signal = tifffile.imread(fetch('Embryo.tif'))
+frequency = 80.11  # MHz; from the XML metadata in the file
+
+print(signal.shape, signal.dtype)
+
+# %%
+# Plot the spatial and histogram averages:
+
+from phasorpy.plot import plot_signal_image
+
+plot_signal_image(signal, axis=0)
+
+# %%
+# Calculate phasor coordinates
+# ----------------------------
+#
+# Phasor coordinates are the real and imaginary components of the complex
+# numbers returned by a real forward Digital Fourier Transform (DFT)
+# of a signal at certain harmonics (multiples of the fundamental frequency),
+# normalized by the mean intensity (the zeroth harmonic).
+# Phasor coordinates are named ``real`` and ``imag`` in the PhasorPy library.
+# In literature and other software, they are also known as
+# :math:`G` and :math:`S` or :math:`a` and :math:`b` (as in :math:`a + bi`).
+#
+# Phasor coordinates at multiple harmonics can be calculated from the signal,
+# a TCSPC histogram in this case. First and second harmonics are calculated
+# in this example. For all harmonics, use ``harmonic='all'``.
+# The histogram samples are in the first dimension (`axis=0`):
+
+from phasorpy.phasor import phasor_from_signal
+
+mean, real, imag = phasor_from_signal(signal, harmonic=[1, 2], axis=0)
+
+# %%
+# Plot the calculated phasor coordinates at the first and second harmonics:
+
+from phasorpy.plot import plot_phasor_image
+
+plot_phasor_image(mean, real, imag, title='Sample')
+
+# %%
+# Calibrate phasor coordinates
+# ----------------------------
+#
+# The signals from time-resolved measurements are convoluted with an
+# instrument response function, causing the phasor-coordinates to be
+# phase-shifted and modulated (scaled) by unknown amounts.
+# The phasor coordinates must therefore be calibrated with coordinates
+# obtained from a reference standard of known lifetime, acquired with
+# the same instrument settings.
+#
+# In this case, a homogeneous solution of Fluorescein with a lifetime of
+# 4.2 ns was imaged.
+#
+# Read the signal of the reference measurement from a file:
+
+reference_signal = tifffile.imread(fetch('Fluorescein_Embryo.tif'))
+
+# %%
+# Calculate phasor coordinates from the measured reference signal at
+# the first and second harmonics:
+
+reference_mean, reference_real, reference_imag = phasor_from_signal(
```

Comment:
That can be interesting. I guess you are thinking of using the `phasor_at_harmonic` to calculate the position of the reference at other harmonics, considering that now we have the harmonic parameter. This can help with performance I think, particularly for multiple harmonics since there is no need to transform all harmonics from the signal.

Although I think if we do this, then we need to change to ask only for the first harmonic (in the harmonic arg). I think the advantage of the current implementation is that both the reference and the sample have the same processing, which can simplify the pipeline (and tutorials, etc). But if this modification reduces the processing time for a pipeline , it is also a good idea. What do you think we should prioritize now?

Reply:
Let's keep this case out of the high level `phasor_calibrate` function. The function is meant for common cases, where a reference is acquired with the same settings as the sample. What might be nice though is to demonstrate in a tutorial that it is possible to calibrate multi-harmonic data with a single-harmonic reference using the lower level functions (`phasor_at_harmonic`, `phasor_transform`, `phasor_from_lifetime`, ...). But that can be added later.

Reply:
Excelent idea. I think for now we can keep this cases out of `phasor_calibrate` but show in the tutorial that it is possible without having to calculate all harmonics from reference. I will try to add it to the tutorial in this PR. In case I see it delays this PR too much, I will postpone it.